Update Support Policy (#88)
sous-chefs/website: content/docs/cookbook-support.md

---
title: 'Cookbook Best Practices'
date: 2020-04-03T00:00:00+00:00
---

Cookbook support is a complicated subject based on many variables, however this document should outline expectations in our cookbooks

> ## Semvar

Reviewer: ```suggestion
## Semver
```

> We follow semvar, this is never optional to our cookbooks, and allows you, our consumers, to understand the change we are making and the potential impact on your environment.

Reviewer: ```suggestion
We follow semver, this is never optional to our cookbooks, and allows you, our consumers, to understand the change we are making and the potential impact on your environment.
```

## Cookbook Support

> We will attempt to support the version of the `chef-client` that chef support unless there is a compelling reason not to, such as:

Reviewer 2: ```suggestion
We will attempt to support the version of the `chef-client` that chef supports unless there is a compelling reason not to, such as:
```

> We will attempt to support the version of the `chef-client` that chef support unless there is a compelling reason not to, such as:

Reviewer 3: ```suggestion
We will attempt to support the version of the Chef Infra Client that Chef Software supports unless there is a compelling reason not to, such as:
```

> - Features we require that are only in a later version of the chef client

Reviewer 3: ```suggestion
- Features we require that are only in a later version of the Chef Infra Client
```

> - Bugs which reqiure a certian version of the chef client

Reviewer 3: ```suggestion
- Bugs which require a certain version of the Chef Infra Client
```

> - Ruby versions which are closely tied to chef version

Reviewer 3: ```suggestion
- Ruby versions which are closely tied to Chef Infra Client versions
```

> However as always we wecome pull-requests to extend our support across multiple versions

Reviewer 2: ```suggestion
However, as always, we welcome pull-requests to extend our support across multiple versions
```
